idb/postgres/internal/encoding/encode.go

```diff
@@ -53,7 +54,52 @@ func convertEvalDelta(evalDelta types.EvalDelta) types.EvalDelta {
 	return evalDelta
 }
 
+func desanitizeNull(str string) string {
+	if str != "" {
+		return strings.ReplaceAll(str, "\\u0000", "\x00")
+	}
+	return str
+}
+
+// SanitizeNullForQuery converts a string into something postgres can store in a jsonb column.
+func SanitizeNullForQuery(str string) string {
+	if str != "" {
+		result := strings.ReplaceAll(str, "\x00", "\\u0000")
+		if len(result) != len(str) {
+			// Escape the escapes if this is going to be used in a query.
+			return strings.ReplaceAll(result, "\\", "\\\\")
```

Comment:
I think you need to replace existing backslashes before adding backslashes (this needs to happen beforre line 67)

Reply:
This is specifically to search for `\u0000`, if you don't escape the slash it doesn't find anything.

I'm simplifying this to be just `return strings.ReplaceAll(str, "\x00", "\\\\u0000").

I think that all other unicode characters would be converted by the json encoder with a single backslash, postgres must convert them into the unicode character.